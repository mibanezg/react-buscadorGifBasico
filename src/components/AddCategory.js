import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {
    const [inputValue, setInputValue] = useState('');
    const handleInputChange= (e)=>{
        setInputValue(e.target.value);
        console.log('handleInputChange llamando');
    }
    const handleSubmit=(e)=>{
        e.preventDefault();
        console.log('handleSubmit llamando' ,inputValue);
    
        if(inputValue.trim().length>2){
            setCategories(cats=>[inputValue,...cats]);
            setInputValue('');
        }    
    }
    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input 
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}
//Para hacer requerido el paso de valores
AddCategory.propTypes={
    setCategories: PropTypes.func.isRequired
}
