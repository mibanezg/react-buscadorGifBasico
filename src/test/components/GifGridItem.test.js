import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas de <GifGridItem/>', () => {

    const title='Titulo';
    const url ='https://localhost';

    const wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('Debe mostrar texto correctamente ', () => {
        
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe tener parrafo con Title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    test('Debe tener una imagen igual al url y props', () => {
        const img = wrapper.find('img');
        expect(img.props().src).toBe(url);
        expect(img.props().alt).toBe(title);
    });

    test('Debe tener animate fadeIn', async () => {
        const div = wrapper.find('div');
        const className = div.prop('className');
        //console.log(div.props().className);
       expect(className.includes('animate__fadeIn')).toBe(true);
    }); 
    
})
