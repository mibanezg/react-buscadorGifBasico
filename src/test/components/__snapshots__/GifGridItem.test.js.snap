// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifGridItem/> Debe mostrar texto correctamente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo"
    src="https://localhost"
  />
  <p>
    Titulo
  </p>
</div>
`;
